lib/laforge/activerecord.rb

```diff
@@ -84,21 +87,42 @@ def update_attribute_source_priority(attribute, source, priority)
     # Returns a list of the entries matching the filters
     def filter_loaded_data_entries(attributes: nil, sources: nil, present: nil)
       list = data_entries.to_a
+      return list if list.blank?
 
-      unless attribute.nil?
+      unless attributes.nil?
         attributes = Array.wrap(attributes).map(&:to_s)
-        list.select! {|data_entry| attributes.include?(data_entry.attribute) }
+        list.select! {|data_entry| attributes.include?(data_entry.attribute_name) }
       end
 
       unless sources.nil?
-        sources = Array.wrap(sources).map {|source| source.id if source.is_a?(ActiveRecord::Base) }
-        list.select! {|data_entry| sources.include?(data_entry.source_id) }
+        source_ids = Array.wrap(sources).map {|source| source_id_from(source) }.compact
```

Comment:
We should enforce a source, so there's no need to ever have blank sources on data entries, so there's no need to compact, and we should also explode in `source_id_from` if we get a blank value.